### PR TITLE
Revert "Add Datadog tracing library (#31294)"

### PR DESCRIPTION
**Pull request by tsenart**
This reverts commit 602d32e97cabf62a015bfce315fe2e21c6936a0a.

## Test plan

Reverts a commit that broken functionality, no test appropriate.

### CODE REVIEW

**Review by kevinwojo — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,8 +29,6 @@ All notable changes to Sourcegraph are documented in this file.
 ### Removed
 
 -
-- The deprecated GraphQL field `SearchResults.resultCount` has been removed in favor of its replacement, `matchCount`. [#31573](https://github.com/sourcegraph/sourcegraph/pull/31573)
-- The deprecated site-config field `UseJaeger` has been removed. Use `"observability.tracing": { "sampling": "all" }` instead [#31294](https://github.com/sourcegraph/sourcegraph/pull/31294/commits/6793220d6cf1200535a2610d79d2dd9e18c67dca)
```
Comment (eseliger) [inline CHANGELOG.md:33]: this seems unrelated?

File: schema/site.schema.json [hunk 2]
```diff
@@ -933,16 +933,15 @@
...
     "observability.tracing": {
       "description": "Controls the settings for distributed tracing.",
       "type": "object",
       "properties": {
-        "type": {
-          "description": "Determines what tracing provider to enable. Supports \"datadog\" or \"opentracing\"",
-          "type": "string",
-          "enum": ["datadog", "opentracing"],
-          "default": "opentracing"
-        },
```
Comment (eseliger) [inline schema/site.schema.json:945]: could this config change be the reason? Cloud maybe is misconfigured?

**Review by eseliger — APPROVED** [review]
approving to unblock

File: schema/site.schema.json (hunk 2 above)
Comment (tsenart) [reply schema/site.schema.json:945]: Maybe, I'll let the @sourcegraph/cloud-devops team follow that lead. Currently I just really need to debug something in production.

File: CHANGELOG.md (hunk 1 above)
Comment (tsenart) [reply CHANGELOG.md:33]: That was part of the original commit.

**Comment by daxmc99** [comment]
I plan to add this back with proper documentation of how to switch between tracing providers. We shouldn't need to completely revert this in the future. We should only need to modify the cloud-config as we did here: 
https://github.com/sourcegraph/deploy-sourcegraph-cloud/pull/15710#pullrequestreview-893737465

Regardless, this change wasn't broadcast widely enough which is why the revert was needed.
